automatic/jenkins/tools/chocolateyinstall.ps1

```diff
@@ -0,0 +1,115 @@
+$ErrorActionPreference = 'Stop'
+
+$ToolsDir = Split-Path -parent $MyInvocation.MyCommand.Definition
+$JenkinsRegistryRoot = "HKLM:\SOFTWARE\Jenkins\InstalledProducts\Jenkins"
+
+# If x86 exists, that is the default installation location for the previous version of Jenkins on an x64 system
+$ProgramFiles = @($env:ProgramFiles, ${env:ProgramFiles(x86)})[[bool]${env:ProgramFiles(x86)}]
+$MsiUpgrade = -not (Test-Path $JenkinsRegistryRoot) -and
+    (Test-Path $ProgramFiles\Jenkins\jenkins.exe) -and
+    [version](Get-ItemProperty $ProgramFiles\Jenkins\jenkins.exe -ErrorAction SilentlyContinue).VersionInfo.ProductVersion -le "2.3.0"
+
+$packageArgs = @{
+    packageName   = $env:ChocolateyPackageName
+    fileType      = 'msi'
+    file          = Join-Path $ToolsDir 'jenkins.msi'
+    silentArgs  = "/qn /norestart /l*v `"$($env:TEMP)\$($env:ChocolateyPackageName).$($env:chocolateyPackageVersion).MsiInstall.log`""
+    validExitCodes= @(0, 3010, 1641)
+    softwareName  = 'Jenkins*'
+}
+
+# Handle Package Parameters
+$PackageParameters = Get-PackageParameters
+foreach ($PotentialParameter in @(
+        "INSTALLDIR"
+        "JENKINS_ROOT"
+        "PORT"
+        "JAVA_HOME"
+        "SERVICE_USERNAME"
+        "SERVICE_PASSWORD"
+    )) {
+    if ($PackageParameters.ContainsKey($PotentialParameter)) {
+        Set-Variable -Name $PotentialParameter -Value $PackageParameters[$PotentialParameter]
```

Comment:
Let's avoid setting new variables when it isn't needed.

the if checks below could use
```
if ($pp['INSTALLDIR'])
```

Instead of setting the variable in the scope.
The indexing is safe to use in this context, even if the user hasn't passed in parameters.

Reply:
To confirm, this method _does not_ set variables in the scope if they have not been passed in? Unless I'm very, very wrong about how Get-PackageParameters is working, here.